File: archived/2023-12/1915e.cpp
#include <bits/stdc++.h>
using namespace std;

// ./run sample.cpp

#define ll long long

int main() {
    int t, n;
    cin >> t;
    while (t--) {
        cin >> n;
        vector<ll> a(n);
        for (ll &x: a) cin >> x;
        set<ll> s;
        s.insert(0);
        bool ans = false;
        bool odd = true;
        ll sum = 0;
        for (ll &x: a) {
            if (odd) sum += x;
            else sum -= x;
            if (s.count(sum)) {
                ans = true;
                break;
            }
            s.insert(sum);
            odd = !odd;
        }
        if (ans) cout << "YES" << endl;
        else cout << "NO" << endl;
    }
}

File: archived/2023-12/1915e.py
t = int(input())
for _ in range(t):
    n = int(input())
    a = [int(i) for i in input().split()]
    s = set([0])
    ans = False
    odd = True
    sum = 0
    for x in a:
        if odd: sum += x
        else: sum -= x
        if sum in s:
            ans = True
            break
        s.add(sum)
        odd = not odd
    print("YES" if ans else "NO")
